src/types/NodeType.ts

```diff
@@ -0,0 +1,347 @@
+/* eslint-disable no-underscore-dangle */
+
+import {
+  GraphQLFieldConfig,
+  GraphQLFieldConfigMap,
+  GraphQLFieldResolver,
+  GraphQLObjectType,
+  GraphQLObjectTypeConfig,
+  GraphQLResolveInfo,
+  GraphQLString,
+  Thunk,
+} from 'graphql';
+import {
+  connectionDefinitions,
+  fromGlobalId,
+  globalIdField,
+  nodeDefinitions,
+} from 'graphql-relay';
+import invariant from 'invariant';
+import camelCase from 'lodash/camelCase';
+import snakeCase from 'lodash/snakeCase';
+import pluralize from 'pluralize';
+import { Class } from 'utility-types';
+
+import Resource from '../resources/Resource';
+import asType from '../utils/asType';
+import resolveThunk from '../utils/resolveThunk';
+
+type Context = any;
+
+// graphql overrides
+
+type ObjStub = { [key: string]: unknown } & {
```

Comment:
`ObjStub` now is an obj with `id` defined. On looking at it now, I could probably do
```typescript
{
  id: string;
  [key: string]: unknown;
}
```

Reply:
I don't think you need the index, an interface with only `id` should be enough, since they aren't exact anyway

Reply:
I think it does need an index. Getting an error here:

```ts
  function getNodeValue(
    obj: ObjStub,
    fieldName: string,
    context: Context,
    info: GraphQLResolveInfo,
  ) {
    if (obj[fieldName] === undefined) {
      return getAsyncNodeValue(obj, fieldName, context, info);
    }

    return obj[fieldName];
  }
```

Given that `fieldname` is any string, `obj[fieldName]` doesn't make sense here.

Reply:
hmm what about: 

```tsx
getNodeValue<TStub extends ObjStub>(obj: TStub, fieldName: keyof TStub)
```

Reply:
getting that annoying
`no index signature with a parameter of type 'string'` when:

```ts
interface ObjStub {
  id: string;
}
```

and 

```ts
function getNodeValue<TStub extends ObjStub>(
    obj: TStub,
    fieldName: string,
    context: Context,
    info: GraphQLResolveInfo,
  ) {
    if (obj[fieldName] === undefined) {
      return getAsyncNodeValue(obj, fieldName, context, info);
    }

    return obj[fieldName];
  }
```

Reply:
missing the `keyof` http://www.typescriptlang.org/play/?ssl=13&ssc=4&pln=1&pc=1#code/JYOwLgpgTgZghgYwgAgPICMBWBlMBXdZAbwChllgATALmQGcwpQBzAbhIF8SSY8QEwwAPYhkzCGAByQyhABqcADZ4IAHgAquAsggAPSCEp00WLegB8ACjLlkQrLU350AGhvkYwCIsqS4AWwhaAGsIAE8hGGQnAjdyAEpidwooy3tMAG1Pb18AiABdZABeEuQ+WU8QCEpE0ltbKAk8KFF0rK8fP0D89nquesb8FuQQPEVFGw4gA